pytensor/tensor/slinalg.py

```diff
@@ -908,6 +908,105 @@ def solve_discrete_are(A, B, Q, R, enforce_Q_symmetric=False) -> TensorVariable:
     )
 
 
+def _largest_common_dtype(tensors: typing.Sequence[TensorVariable]) -> np.dtype:
+    return reduce(lambda l, r: np.promote_types(l, r), [x.dtype for x in tensors])
+
+
+class BaseBlockDiagonal(Op):
+    __props__ = ("gufunc_signature", "n_inputs")
```

Comment:
Not all variables need to be props. Usually we make props the ones needed to initialize the Op (for instance the inplace rewrite).

The real contract for props, however, is that if two Ops have the same type, same inputs and same props, then they can be considered equivalent and merged. For this use case we actually don't need any props.

Still for the use case of more easily reconstructing an Op I would include n_inputs only. This is also used for the automatic `__repr__`

Another reason not to include the signature is that it's always a prop of the Blockwise and would show twice in the dprint when you have a Blockwise BlockDiagonal 

```suggestion
    __props__ = ("n_inputs")
```